clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -384,9 +418,12 @@
                                                                         0),
                                                               new Field("responses",
                                                                       new ArrayOf(FETCH_RESPONSE_TOPIC_V0)));
+    // Fetch response V2 is the same as fetch response V1. The version number is bumped up to indicate that the Message
```

Comment:
Fetch response v2 is actually different from v1 since the message format is different.